### PR TITLE
doc: improve metadata parsing so that titles can be reliably extracted

### PR DESCRIPTION
**Pull request by dhowden**
Need to run a few more tests, posting for reference.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -23,28 +25,43 @@ func ConvertDoc(r io.Reader) (string, map[string]string, error) {
 	// Meta data
 	mc := make(chan map[string]string, 1)
 	go func() {
+		defer func() {
+			if e := recover(); e != nil {
+				log.Printf("panic when reading doc format: %v", e)
```
Comment (jonathaningram) [inline doc.go:30]: `recovered panic`?

**Review by jonathaningram — COMMENTED** [review]
Guess tests would be nice, not sure how easy they are to add. I know there are some that do test things. I think the GH action installs some binaries so it may even run tests against the real tool.

**Review by justinkoke — APPROVED** [review]

File: doc.go (hunk 1 above)
Comment (dhowden) [reply doc.go:30]: Yeah, scanned over the metadata extraction code and there seems to be a lot of places for malformed documents to cause panics.

Just putting this here to stop it failing the doc conversion.  Noting previously we would just ignore fatal errors calling out to wvSummary, so this seemed equivalent.

Comment (jonathaningram) [reply doc.go:30]: I meant change log message so we don't freak out and think it panicked when reading logs.

**Pull request merged**